src/api/services/base.py

```diff
@@ -1,28 +1,34 @@
 import abc
 
 from fastapi import Depends
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from src.core.db.repository import ContentRepository
 
 
 class ContentService(abc.ABC):
     """Абстрактный класс для контента."""
 
-    def __init__(self, repository: any = Depends()):
+    def __init__(self, repository: ContentRepository, session: AsyncSession):
         self._repository = repository
+        self._session = session
 
     async def actualize_objects(self, objects: list[any], model_class: any) -> None:
         to_create, to_update = [], []
         ids = [obj.id for obj in objects]
-        await self._repository.archive_by_ids(ids)
-        already_have = await self._repository.get_all_non_archived_and_by_ids(ids)
-        for obj in objects:
-            if obj.id not in already_have:
-                to_create.append(model_class(**obj.dict(), is_archived=False))
-            else:
-                to_update.append({**obj.dict(), "is_archived": False})
-        if to_create:
-            await self._repository.create_all(to_create)
-        if to_update:
-            await self._repository.update_all(to_update)
+        async with self._session as session:
```

Comment:
Вроде тут должно быть так:  
```suggestion
        async with self._session.begin() as session:
```